kernel/kernel-util/src/main/java/org/sakaiproject/util/EmailNotification.java

```diff
@@ -645,6 +645,24 @@ protected int getOption(User user, String notificationId, String resourceFilter,
 		String type = getType(resourceFilter);
 		if (type != null)
 		{
+            // First, check the overrides.
```

Comment:
I did a new branch so I could to a single commit for the issue. The Moodle community enforce that although I'm ambivalent about it, to be honest. The LInux kernel guidelines state that you should split patches up into bits if there are lots of changes, to make it easier to review, so they're okay with multiple commits.